### ISSUE TITLE
Bug in `LightningModule.load_from_checkpoint`

### ISSUE THREAD

**Issue by acturner** [PyTorchLightning/pytorch-lightning#2769]
Suppose you have a class `Model(LightningModule)` (a `LightningModule` subclass) and as parameters it takes
`params: argparse.Namespace` (without a default value), along with other keyword arguments which might take default arguments. Loading from a checkpoint

```python
model = Model(params=params, **kwargs)
...  # saves a checkpoint
model = Model.load_from_checkpoint(params=params, **kwargs)  
# same behavior arises if params is passed as positional argument
```

will throw the error `TypeError: __init__() got multiple values for argument 'params'`.
This comes from the fact that in line `205` in `pytorch_lightning/core/saving.py`

```python
model = cls(*cls_args, **cls_kwargs)
```

when inspected, `cls_args` is `({'params': params},)` and `cls_kwargs` is, of course, `{'params': params}`.
This stems from the fact in lines `189-196` (in `saving.py`):

```python
            if args_name == 'kwargs':
                # in case the class cannot take any extra argument filter only the possible
                cls_kwargs.update(**model_args)
            elif args_name:
                if args_name in cls_init_args_name:
                    cls_kwargs.update({args_name: model_args})
            else:
                cls_args = (model_args,) + cls_args
```

the `else` clause is getting called and `model_args` contains a copy of kwargs!

To be honest, it's very unclear to me exactly what `model_args` is supposed to be here, but what is clear is that
this block needs more checking so that we don't just stick copies of items in `cls_kwargs` into `cls_args`. Just doing
something silly like replacing the `else` clause with

```python
from typing import Sequence
...
             elif isinstance(model_args, Sequence):
                cls_args = tuple(model_args) + cls_args
```

fixes the problem for me, but as I said, I'm unclear on exactly how model_args is being used, so I'm sure there is a better
solution.

**Comment by aturner-sesco** [comment]
@Borda any updates on this front?

**Comment by acturner** [comment]
@Borda any updates on this?

**Comment by Borda** [comment]
@acturner it is fixed by #2776 mind try...
